packages/core/content-type-builder/admin/src/components/FormModal/reducer.js

```diff
@@ -220,6 +221,46 @@ const reducer = (state = initialState, action) =>
         draftState.initialData = action.data;
         break;
       }
+      case actions.SET_CUSTOM_FIELD_DATA_SCHEMA: {
+        const { customField, isEditing, modifiedDataToSetForEditing, options = {} } = action;
+
+        if (isEditing) {
+          draftState.modifiedData = modifiedDataToSetForEditing;
+          draftState.initialData = modifiedDataToSetForEditing;
+
+          break;
+        }
+
+        draftState.modifiedData = { ...options, type: customField.type };
+
+        if (customField?.options?.defaults) {
+          // Set default values
+          Object.entries(customField.options.defaults).forEach(([keyPath, value]) => {
+            const allowedRootLevelOptions = [
+              'min',
+              'max',
+              'minLength',
+              'maxLength',
+              'required',
+              'regex',
+              'enum',
+              'unique',
+              'private',
+              'default',
+            ];
+
+            if (allowedRootLevelOptions.includes(keyPath) || keyPath.startsWith('options')) {
+              // Set the value on the root or options object
+              _.set(draftState.modifiedData, keyPath, value);
+            } else {
+              // Set everything else on the options object
+              _.set(draftState.modifiedData, ['options', keyPath], value);
```

Comment:
I think this means there are 2 ways to do the same thing, which I'm not a fan of (`options.customStuff` ends up the same as `customStuff` directly)

I know it's for convenience, but personally I'd prefer to throw an explicit error message (so probably at the registry level, not here)

Reply:
I was thinking more as a fallback, we would only document the way we want it to be done. I think an error at the registry level makes more sense though.